### PR TITLE
Updated CHANGELOG for ROCm 6.1

### PR DESCRIPTION
**Pull request by causten**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,81 @@
...
+* Handle all slice input variations
+* Add scales attribute parse in upsample for older opset versions
+* Added support for uneven Split operations
+* Improved unit testing to run in python virtual environments
+
+### Fixes
+
+* Fixed outstanding issues in autogenerated documentation
+* Update model zoo paths for examples
+* Fixed compute_shape of `mlir_op` and the `fused_concat` operators
```
Comment (umangyadav) [inline CHANGELOG.md:56]: AFAIK this wasn't broken in rocm-6.0 Therefore doesn't need to be part of "fixes"

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,81 @@
...
+* Improved performance by always using rocMLIR for quantized convolution
+* Improved group convolutions by using rocMLIR
+* Improved accuracy of fp16 models
+* ScatterElements unsupported reduction
+* Added concat fusions
+* Improved INT8 support to include UINT8
+* Allow reshape ops between dq and quant_op
+* Improve dpp reductions on navi
+* Have the accuracy checker print the whole final buffer
+* Simplify constant of shape and  constant variable input slices
```
Comment (CharlieL7) [inline CHANGELOG.md:42]: Can reword this to: Added support for handling dynamic Slice and ConstantOfShape ONNX operators.

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,81 @@
...
+* Add support for the dilations attribute to Pooling ops
+* Simplify dynamic reshape
```
Comment (CharlieL7) [inline CHANGELOG.md:44]: Not a user-facing change. AFAIK, was a result of adding `reshape_lazy` and having `reshape` do a copy.

**Review by umangyadav — APPROVED** [review]
